### PR TITLE
Clone pod features, fixes, tweaks

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
rscadd: Clone pods now notify the medical channel on a successful clone.
They also notify the medical channel if the clone dies or is horribly
mutilated; but not if it is ejected early by authorised medical
personnel.
rscadd: Cloning pods now are more capable of cloning humanoid species
that do not breathe.
/:cl:
- Cleaned up cloning.dm, removed a bunch of unused code, use TRUE and
  FALSES, removed trailing returns, put some more colourful messages in.
- Small amounts of Bicardine are added to the clone to cope with any
  NOBREATH crit damage.
- The `grab_ghost_when` var has been added, which will change when we
  actually grab the ghost and stick it in the new body. By default it is
  unchanged to when the clone is freshly made, but can be changed to
  "mature" which is when the clone is ejected.

### CODE REVIEW

File: code/game/machinery/cloning.dm [hunk 1]
```diff
@@ -173,74 +168,78 @@
...
-			src.locked = 0
-			src.go_out()
-			src.connected_message("Clone Rejected: Deceased.")
-			return
+	else if((occupant) && (occupant.loc == src))
+		if((occupant.stat == DEAD) || (occupant.suiciding) || !occupant.key || occupant.hellbound)  //Autoeject corpses and suiciding dudes.
+			locked = FALSE
+			go_out()
+			connected_message("Clone Rejected: Deceased.")
+			speak("The cloning of <b>[occupant]</b> has been aborted due to \
```
Comment (optimumtact) [inline code/game/machinery/cloning.dm]: speak ???! ?!?!?! ?!

**Comment by PKPenguin321** [comment]
>rscadd: Clone pods now notify the medical channel on a successful clone.
They also notify the medical channel if the clone dies or is horribly
mutilated; but not if it is ejected early by authorised medical
personnel.

hugbox

**Comment by LatD** [comment]
Can you somehow make clones to not breathe air from outside the pod while they're inside? Put a magical gas mask on them or something, I have no idea how to implement this. It would just make sense that they're not dying inside the pod while cloning if the room is filled with plasma. This would also be the fix for #11213

**Comment by coiax** [comment]
I have an idea

**Comment by optimumtact** [comment]
mergeable?

**Comment by coiax** [comment]
Yes, I will make another PR for my idea.

**Comment by lordpidey** [comment]
>Small amounts of Bicardine are added to the clone to cope with any NOBREATH crit damage.

Uhh...  Bicardine takes care of brute damage, not oxyloss.

**Comment by coiax** [comment]
@lordpidey NOBREATH species cannot take oxyloss. Instead, if they're in crit, they take low amounts of brute damage.

**Pull request merged**